### PR TITLE
Updated Dart sdk to v4.0.0 Catnip

### PR DESCRIPTION
**Pull request by novalisdenahi**
### Description

Update dart sdk to v4.0.0 (catnip)

### Trello card

[Trello link](https://trello.com/c/oWIaQxAh/337-catnip-dart)

### Notes for QA

The SDK docs updated and the config-v2-sdk-support-table.md

### Requirement checklist

- [x] I have validated my changes on a test/local environment.
- [ ] I have checked the SNYK/Dependabot reports and applied the suggested changes.
- [ ] (Optional) I have updated outdated packages.

### CODE REVIEW

**Review by sigewuzhere — APPROVED** [review]

**Review by adams85 — CHANGES_REQUESTED** [review]
Could you also adapt and add
* this part to the `getValue` section: https://github.com/configcat/docs/blob/f850e29263c8d55df752fb022c734a678eafda03/website/docs/sdk-reference/dotnet.md?plain=1#L156-L185
* this part to the `getValueDetails` section: https://github.com/configcat/docs/blob/f850e29263c8d55df752fb022c734a678eafda03/website/docs/sdk-reference/dotnet.md?plain=1#L207-L210

**Review by adams85 — APPROVED** [review]

**Review by adams85 — CHANGES_REQUESTED** [review]
In the meantime, we merged the Config V2 docs, which includes some formatting/spelling consistency improvements in the SDK references. See e.g. https://github.com/configcat/docs/commit/6f8d20ce9721d39cbba7a43541aa37613bfdcb94#diff-f1a725410bdd352c87347015b639baff90bc65cc7a1c8b38537e04ae5f8e421e

Please align the PR with these changes.

**Review by z4kn4fein — DISMISSED** [review]

File: website/docs/sdk-reference/dart.mdx [hunk 1]
```diff
@@ -155,6 +155,39 @@ final value = await client.getValue(
...
+### Setting type mapping {#setting-type-mapping}
+
+| Setting Kind   | Type parameter `T` |
+| -------------- |--------------------|
+| On/Off Toggle  | `bool`/`bool?`     |
+| Text           | `String`/`String?` |
+| Whole Number   | `int`/`int?`       |
+| Decimal Number | `double`/`double?` |
+
+In addition to the types mentioned above, you also have the option to provide `dynamic` for the type parameter regardless of the setting kind.
```
Comment (adams85) [inline website/docs/sdk-reference/dart.mdx:172]:
```suggestion
In addition to the types mentioned above, you also have the option to provide `object`, `object?` or `dynamic` for the type parameter regardless of the setting kind.
```

File: website/docs/sdk-reference/dart.mdx [hunk 2]
```diff
@@ -173,18 +206,23 @@ final details = await client.getValueDetails(
...
+| `value`                     | `bool` / `String` / `int` / `double` | The evaluated value of the feature flag or setting.                                       |
+| `key`                       | `String`                             | The key of the evaluated feature flag or setting.                                         |
+| `isDefaultValue`            | `bool`                               | True when the default value passed to getValueDetails() is returned due to an error.      |
+| `error`                     | `String?`                            | In case of an error, this field contains the error message.                               |
+| `user`                      | `ConfigCatUser?`                     | The User Object that was used for evaluation.                                             |
+| `matchedPercentageOption`   | `PercentageOption?`                  | If the evaluation was based on a percentage rule, this field contains that specific rule. |
+| `matchedTargetingRule`      | `TargetingRule?`                     | If the evaluation was based on a Targeting Rule, this field contains that specific rule.  |
```
Comment (adams85) [inline website/docs/sdk-reference/dart.mdx]:
```suggestion
| `matchedPercentageOption`   | `PercentageOption?`                  | The Percentage Option (if any) that was used to select the evaluated value. |
| `matchedTargetingRule`      | `TargetingRule?`                     | The Targeting Rule (if any) that matched during the evaluation and was used to return the evaluated value.  |
```

File: website/docs/sdk-reference/dart.mdx [hunk 3]
```diff
@@ -173,18 +206,23 @@ final details = await client.getValueDetails(
...
-| `matchedEvaluationRule`           | `RolloutRule?`                       | If the evaluation was based on a Targeting Rule, this field contains that specific rule.  |
-| `fetchTime`                       | `DateTime`                           | The last download time of the current config.                                             |
+| Field                       | Type                                 | Description                                                                               |
+|-----------------------------|--------------------------------------|-------------------------------------------------------------------------------------------|
+| `value`                     | `bool` / `String` / `int` / `double` | The evaluated value of the feature flag or setting.                                       |
+| `key`                       | `String`                             | The key of the evaluated feature flag or setting.                                         |
+| `isDefaultValue`            | `bool`                               | True when the default value passed to getValueDetails() is returned due to an error.      |
```
Comment (adams85) [inline website/docs/sdk-reference/dart.mdx]:
```suggestion
| `isDefaultValue`            | `bool`                               | True when the default value passed to `getValueDetails()` is returned due to an error.      |
```

File: website/docs/sdk-reference/dart.mdx [hunk 4]
```diff
@@ -218,6 +256,48 @@ final user = ConfigCatUser(
...
+
+**Date time-based comparators** (BEFORE / AFTER)
+* accept `DateTime` values, which are automatically converted to a second-based Unix timestamp,
+* accept `double` values representing a second-based Unix timestamp and all other numeric values which can safely be converted to `Double`,
+* accept `String` values containing a properly formatted, valid `double` value,
+* all other values are considered invalid (a warning will be logged and the currently evaluated targeting rule will be skipped).
+
+**String array-based comparators** (ARRAY CONTAINS ANY OF / ARRAY NOT CONTAINS ANY OF)
+* accept arrays and list of `String`,
+* accept `String` values containing a valid JSON string which can be deserialized to an array of `String`,
```
Comment (adams85) [inline website/docs/sdk-reference/dart.mdx:298]: https://dart.dev/language/collections#lists

> Perhaps the most common collection in nearly every programming language is the array, or ordered group of objects. In Dart, arrays are [List](https://api.dart.dev/stable/dart-core/List-class.html) objects, so most people just call them lists.

```suggestion
* accept lists or sets of `String`,
* accept `String` values containing a valid JSON string which can be deserialized to a list of `String`,
```

**Review by adams85 — CHANGES_REQUESTED** [review]

**Pull request merged**